### ISSUE TITLE
1.5.0: The dependency target "update_helpfiles" of target "dmidiplayer" does not exist

### ISSUE THREAD

**Issue by yurivict** [pedrolcl/dmidiplayer#6]
```
CMake Error at CMakeLists.txt:208 (add_dependencies):
  The dependency target "update_helpfiles" of target "dmidiplayer" does not
  exist.


CMake Error in CMakeLists.txt:
  The dependency target "update_helpfiles" of target "dmidiplayer_autogen"
  does not exist.
```

OS: FreeBSD 13

**Comment by pedrolcl** [comment]
Hi Yuri,

On miércoles, 25 de agosto de 2021 6:20:22 (CEST) ***@***.*** wrote:
> ```
> CMake Error at CMakeLists.txt:208 (add_dependencies):
>   The dependency target "update_helpfiles" of target "dmidiplayer" does
> not exist.
> 
> 
> CMake Error in CMakeLists.txt:
>   The dependency target "update_helpfiles" of target
> "dmidiplayer_autogen" does not exist.
> ```
> 
> OS: FreeBSD 13

This release uses Pandoc at build time to convert from Markdown documents 
into HTML files that will be included in the executable as embedded 
resources. In previous releases, Pandoc was already used for conversion 
from Mardown to Groff when generating the man page. Both the HTML files and 
the man page are included in the source tarball, so Pandoc should be 
optional, as cmake reports only a warning at configuration time when it is 
not found.

There are two possible solutions: 

* Either you can install Pandoc before configuration: 
https://www.freshports.org/textproc/hs-pandoc/

* Or you can apply this patch (it will be included in the next release):
https://sourceforge.net/p/dmidiplayer/code/ci/
4b01d7911828bd770cdc6d59a56d44fd509b5b94/tree/CMakeLists.txt?
diff=8b53808989c84fc5cb3510ab357bed7e8b3eb38c

Beware that if you execute "make clean" in the build directory, the HTML 
files will be erased, and then you will need Pandoc to create them again. 
Also, if you modify the source Markdown documents, then you will also need 
Pandoc to generate the output files.

You may also have noticed that the Russian translation is quite outdated, 
and the Russian help document is a machine translation, probably of very 
low quality.  If you can help, or you know somebody who can help with the 
translations, I would be grateful.

Regards,
Pedro

**Comment by yurivict** [comment]
Instead of ```if (EXISTS ${PANDOC_EXECUTABLE})``` shouldn't it be ```if (BUILD_DOCS)``` ?

**Comment by pedrolcl** [comment]
> Instead of `if (EXISTS ${PANDOC_EXECUTABLE})` shouldn't it be `if (BUILD_DOCS)` ?

Yes, you may do that. But if you look at the definition of `BUILD_DOCS`:

~~~
option(BUILD_DOCS "Build man pages" ON)
~~~

So, a proper fix would be to redefine the `BUILD_DOCS` option to reference any other Markdown sources in addition to man pages. Also, if you search the usages of `BUILD_DOCS`, you will find that it is not used anywhere else.  In PandocMacros.cmake, when the test for Pandoc fails, it should report a warning only if `BUILD_DOCS` is OFF, reporting a fatal error otherwise.

**Comment by pedrolcl** [comment]
Fixed in v1.5.1 and released
